crates/miden-objects/src/block/block_inputs.rs

```diff
@@ -128,3 +131,35 @@ impl BlockInputs {
         &mut self.account_witnesses
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for BlockInputs {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.prev_block_header.write_into(target);
+        self.partial_blockchain.write_into(target);
+        self.account_witnesses.write_into(target);
+        self.nullifier_witnesses.write_into(target);
+        self.unauthenticated_note_proofs.write_into(target);
+    }
```

Comment:
I like deconstructing in these type of situations so that one is forced to revisit this if you add a new field


```suggestion
    fn write_into<W: ByteWriter>(&self, target: &mut W) {
        let Self {
            prev_block_header,
            partial_blockchain,
            account_witnesses,
            nullifier_witnesses,
            unauthenticated_note_proofs,
        } = self;
        
        prev_block_header.write_into(target);
        partial_blockchain.write_into(target);
        account_witnesses.write_into(target);
        nullifier_witnesses.write_into(target);
        unauthenticated_note_proofs.write_into(target);
    }
```